ec2/spark_ec2.py

```diff
@@ -1311,7 +1314,10 @@ def real_main():
         sys.exit(1)
 
     try:
-        conn = ec2.connect_to_region(opts.region)
+        if opts.profile != "default":
```

Comment:
What if I wanted to have a profile that's named "default"? I think it would better to have `None` as the default and to replace this with an `is not None` check.